rtl/ompt-tsan.cpp

```diff
@@ -142,36 +142,42 @@ extern "C" {
     fptr = (void (*)(const char *, int, const volatile void *))dlsym(RTLD_DEFAULT, "AnnotateHappensAfter");
     (*fptr)(file,line,cv);
   }
+
   static void AnnotateHappensBefore(const char *file, int line, const volatile void *cv){
     void (*fptr)(const char *, int, const volatile void *);
 
     fptr = (void (*)(const char *, int, const volatile void *))dlsym(RTLD_DEFAULT, "AnnotateHappensBefore");
     (*fptr)(file,line,cv);
   }
+
   static void AnnotateIgnoreWritesBegin(const char *file, int line){
     void (*fptr)(const char *, int);
 
     fptr = (void (*)(const char *, int))dlsym(RTLD_DEFAULT, "AnnotateIgnoreWritesBegin");
     (*fptr)(file,line);
   }
+
   static void AnnotateIgnoreWritesEnd(const char *file, int line){
     void (*fptr)(const char *, int);
 
     fptr = (void (*)(const char *, int))dlsym(RTLD_DEFAULT, "AnnotateIgnoreWritesEnd");
     (*fptr)(file,line);
   }
+
   static void AnnotateNewMemory(const char *file, int line, const volatile void *cv, size_t size){
     void (*fptr)(const char *, int, const volatile void *,size_t);
 
     fptr = (void (*)(const char *, int, const volatile void *,size_t))dlsym(RTLD_DEFAULT, "AnnotateNewMemory");
     (*fptr)(file,line,cv,size);
   }
+
   static int RunningOnValgrind(){
     int (*fptr)();
 
     fptr = (int (*)())dlsym(RTLD_DEFAULT, "RunningOnValgrind");
-    if (fptr==NULL || fptr != RunningOnValgrind)
+    if (fptr == NULL)
```

Comment:
@jprotze `fptr` pointer is `NULL` when `-fsanitize=thread` is not used, `not NULL` otherwise. So the only way to activate Archer is to keep only this check. Let me know what you think.

Reply:
The reasoning behind the extra check `fptr != RunningOnValgrind` is, that the dynamic loader might find the function RunningOnValgrind in ompt-tsan. In that case, we can also assume that there is no tsan in the application.